internal/msgqueue/v1/rabbitmq/rabbitmq.go

```diff
@@ -152,7 +152,7 @@ func WithMessageRejection(enabled bool, maxDeathCount int) MessageQueueImplOpt {
 }
 
 // New creates a new MessageQueueImpl.
-func New(fs ...MessageQueueImplOpt) (func() error, *MessageQueueImpl) {
+func New(fs ...MessageQueueImplOpt) (func() error, *MessageQueueImpl, error) {
```

Comment:
Added an error to the return values because we weren't casing on if `initQueue` failed in this constructor which gets us into an undefined state (leading to a panic)